[Website] Update deps

bkmgit opened the pull request.

kou commented on package.json:
```diff
@@ -18,10 +18,8 @@
-    "jquery": "^3.5.1",
-    "popper.js": "^1.16.1",
-    "webpack": "^5.76.0",
+    "bootstrap": "^5.3.2",
```
Updating Bootstrap 4 -> 5 may break design. Could you check it?

bkmgit replied:
It seems ok. Most of the documentation is built in the other Arrow repositories.

kou replied:
Could you share what you checked?
This PR also updates `jekyll-feed`. So we need to check at least https://arrow.apache.org/feed.xml too.

kou replied:
Providing preview is also helpful.
See also: https://github.com/apache/arrow-site#forks

bkmgit replied:
The current preview workflow at https://github.com/apache/arrow-site/blob/main/.github/workflows/deploy.yml fails with newer Jekyll as the one used by GitHub is older.

The current site can be previewed at https://bkmgit.github.io/arrow-site - without changes

Associated workflow is at:
https://github.com/bkmgit/arrow-site/blob/bkmgit-patch-1/.github/workflows/preview.yml
Can make a pull request to add this so that creating preview sites is easier.
- It does not download the crossbow artifacts, though can add these if they are helpful
- One would need to change permissions to enable deployment, see https://stackoverflow.com/questions/76937061/branch-master-is-not-allowed-to-deploy-to-github-pages-due-to-environment-prot
- If the preview site is created within a subfolder of https://apache.github.io/arrow-site/ say https://apache.github.io/arrow-site/preview/2024-01-06-12-30 then a cron job can be used to remove the preview but having the preview hosted from the origin of the pull request seems sufficient.  Just need to document what to do.

kou replied:
> The current site can be previewed at https://bkmgit.github.io/arrow-site - without changes

We want a preview "with" these changes for this PR.

> Associated workflow is at:
> https://github.com/bkmgit/arrow-site/blob/bkmgit-patch-1/.github/workflows/preview.yml
> Can make a pull request to add this so that creating preview sites is easier.

Can we extend our existing `deploy.yml` workflow instead of creating a new workflow?
It already has some codes for preview such as https://github.com/apache/arrow-site/blob/main/.github/workflows/deploy.yml#L63 .

> If the preview site is created within a subfolder of https://apache.github.io/arrow-site/

We want to create a preview in a fork repository such as https://github.com/bkmgit/arrow-site/ not apache/arrow-site.

bkmgit replied:
Modified existing `deploy.yml`. 
- https://github.com/apache/arrow-site/pull/453 uses baseurl
From https://mademistakes.com/mastering-jekyll/site-url-baseurl/ probably using baseurl is good. Will come back to this once working preview is merged.

bkmgit replied:
Preview is available at:
https://bkmgit.github.io/arrow-site/

Rss feed validation gives a few errors, but these are also present on current rss feed:
https://validator.w3.org/feed/check.cgi?url=https%3A%2F%2Fbkmgit.github.io%2Farrow-site%2Ffeed.xm

> 
>    Recommendations
> 
> This feed is valid, but interoperability with the widest range of feed readers could be improved by implementing the following recommendations.
> 
>     [line 60](https://validator.w3.org/feed/check.cgi?url=https%3A%2F%2Fbkmgit.github.io%2Farrow-site%2Ffeed.xml#l60), column 1859: Two entries with the same value for atom:updated: 2023-11-09T00:00:00-05:00 [[help](https://validator.w3.org/feed/docs/warning/DuplicateUpdated.html)]
> 
>         ... ished><updated>2023-11-09T00:00:00-05:00</updated><id>https://bkmgit.git ...
>                                                      ^
> 
>     [line 1090](https://validator.w3.org/feed/check.cgi?url=https%3A%2F%2Fbkmgit.github.io%2Farrow-site%2Ffeed.xml#l1090), column 0: Invalid HTML: Unexpected '-' after '--' found in comment. [[help](https://validator.w3.org/feed/docs/warning/NotHtml.html)]
> 
>         implementation, see the latest <a href="https://github.com/apache/arrow-rs/t ...

kou replied:
The former isn't problem because we have 2 posts on 2023-11-09.

The latter will be fixed by the following. Could you send a separated PR?

```diff
diff --git a/_posts/2023-08-05-datafusion_fast_grouping.md b/_posts/2023-08-05-datafusion_fast_grouping.md
index f8b5a50e5ef..57fc1b70d87 100644
--- a/_posts/2023-08-05-datafusion_fast_grouping.md
+++ b/_posts/2023-08-05-datafusion_fast_grouping.md
@@ -25,7 +25,7 @@ limitations under the License.
 {% endcomment %}
 -->
 
-<!--- Converted from Google Docs using https://www.buymeacoffee.com/docstomarkdown --->
+<!-- Converted from Google Docs using https://www.buymeacoffee.com/docstomarkdown -->
 
 ## Aggregating Millions of Groups Fast in Apache Arrow DataFusion
 
```

kou replied:
The preview site has `<meta name="generator" content="Jekyll v4.2.0" />`.
Is it really generated with Jekyll 4.3.3?

bkmgit replied:
PR to fix rss https://github.com/apache/arrow-site/pull/461

Thanks, had served pages from incorrect directory. Needed to add base_url to config to
get correct urls in the preview.  Using environment variables seems to repeat the base_url
so have put this in the extra_config.yml file

Preview at https://bkmgit.github.io/arrow-site/

bkmgit replied:
Need to check navbar, other things seem to work.

bkmgit replied:
@kou Dropdowns work now. Let me know if anything else is needed.

alamb commented on Gemfile:
```diff
@@ -18,13 +18,13 @@
 ruby RUBY_VERSION
 
-gem "jekyll", "4.2.0"
+gem "jekyll", "4.3.3"
```
I tested updating jekyl locally and found it let me build the site -- see https://github.com/apache/arrow-site/pull/451 for a PR with just that change

jorisvandenbossche commented:
A few things I noticed on the preview:

1\. The labels on blog posts have a wrong color that make them invisible (https://bkmgit.github.io/arrow-site/blog/2023/12/18/14.0.2-release/):

![image](https://github.com/apache/arrow-site/assets/1020496/70541a97-5ded-4b7b-9bb6-f1e5fa89675c)

versus (https://arrow.apache.org/blog/2023/12/18/14.0.2-release/):

![image](https://github.com/apache/arrow-site/assets/1020496/a11d2f9f-bc8d-4bfd-b145-56b330b5c7e0)

2\. The last dropdown (ASF links) is too close to the border of the window such that it slightly falls off:

![image](https://github.com/apache/arrow-site/assets/1020496/0f814ea7-edfb-46f8-bc0a-1f4ec9000d48)

versus:

![image](https://github.com/apache/arrow-site/assets/1020496/9972a6ab-b9ce-4056-b574-29de47021655)

It seems that the navbar no longer has left/right padding in the updated bootstrap (also the logo on the left side is closer to the side)

3\. On the landing page, the top bar is now a slightly darker grey than the the background of the big logo banner, while before this was they same color (so you didn't visually distinguish the navbar on the landing page from the big banner). Is that change intentional?

4\. In the mobile layout, the navigation doesn't work (clicking the "hamburger" menu doesn't do anything for me):
![Screenshot from 2024-01-09 09-17-40](https://github.com/apache/arrow-site/assets/1020496/d7dfa3ff-1380-44dc-b72d-21fc19c4d578)

bkmgit commented:
@jorisvandenbossche Thanks for the feedback.

1. Updated tags on blog  template, so they are now visible
2. Updated the tag and added a margin, hopefully ok now
3. The colors in bootstrap have changed to improve contrast. Changed the background of the big banner to match. Suggestions for other colors are welcome.
4. Updated a tag that was not previously updated. Menu works on small screens.

Other suggestions are welcome. Site can be previewed at:
https://bkmgit.github.io/arrow-site/

bkmgit commented:
@jorisvandenbossche In the upgraded version all hyperlinks are underlined, otherwise it seems quite similar. Can figure out how to remove the underlines in hyperlinks.
